pkg/odo/cli/service/create.go

```diff
@@ -237,31 +252,40 @@ func (o *ServiceCreateOptions) Validate() (err error) {
 			if err != nil {
 				return err
 			}
-			// var jsonCR map[string]interface{}
-			err = yaml.Unmarshal(fileContents, &o.CustomResourceDefinition)
+
+			err = yaml.Unmarshal(fileContents, &d.OriginalCRD)
 			if err != nil {
 				return err
 			}
 
 			// Check if the operator and the CR exist on cluster
-			o.CustomResource = o.CustomResourceDefinition["kind"].(string)
-			csvs, err := o.KClient.GetClusterServiceVersionList()
+			var csv olm.ClusterServiceVersion
+			o.CustomResource, csv, err = svc.GetCSV(o.KClient, d.OriginalCRD)
 			if err != nil {
 				return err
 			}
 
-			csv, err := doesCRExist(o.CustomResource, csvs)
+			// all is well, let's populate the fields required for creating operator backed service
+			o.group, o.version, o.resource, err = svc.GetGVRFromOperator(csv, o.CustomResource)
 			if err != nil {
-				return fmt.Errorf("Could not find specified service/custom resource: %s\nPlease check the \"kind\" field in the yaml (it's case-sensitive)", o.CustomResource)
+				return err
 			}
 
-			// all is well, let's populate the fields required for creating operator backed service
-			o.group, o.version = groupVersionALMExample(o.CustomResourceDefinition)
-			o.resource = resourceFromCSV(csv, o.CustomResource)
-			o.ServiceName, err = serviceNameFromCRD(o.CustomResourceDefinition, o.ServiceName)
-			return err
-		}
-		if o.CustomResource != "" {
+			o.ServiceName, err = d.getServiceNameFromCRD()
```

Comment:
shouldn't we use the user provided name in case of creation from a file?

Reply:
User has the option to do that when they provide a file. If they're editing other stuff in the file, expecting them to edit the name is not a big ask, IMO. Plus, the `--from-file` feature is not here to stay. It's gonna go away once we have #2785.

Reply:
Added this to avoid having long discussion. Added tests for it as well.